python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,464 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Provides support to run compiled networks both locally and remotely.
+"""
+import json
+import logging
+import os
+import tarfile
+import tempfile
+
+import numpy as np
+import tvm
+from tvm import rpc
+from tvm.autotvm.measure import request_remote
+from tvm.contrib import graph_runtime as runtime
+from tvm.contrib.debugger import debug_runtime
+
+from . import common
+from .common import TVMCException
+from .main import register_parser
+
+
+# pylint: disable=invalid-name
+logger = logging.getLogger("TVMC")
+
+
+@register_parser
+def add_run_parser(subparsers):
+    """ Include parser for 'run' subcommand """
+
+    parser = subparsers.add_parser("run", help="run a compiled module")
+    parser.set_defaults(func=drive_run)
+
+    # TODO --device needs to be extended and tested to support other targets,
+    #      like 'cl', 'webgpu', etc (@leandron)
+    parser.add_argument(
+        "--device",
+        choices=["cpu", "gpu"],
+        default="cpu",
+        help="target device to run the compiled module. Defaults to 'cpu'",
+    )
+    parser.add_argument(
+        "--fill-mode",
+        choices=["zeros", "ones", "random"],
+        default="random",
+        help="fill all input tensors with values. In case --inputs/-i is provided, "
+        "they will take precedence over --fill-mode. Any remaining inputs will be "
+        "filled using the chosen fill mode. Defaults to 'random'",
+    )
+    parser.add_argument("-i", "--inputs", help="path to the .npz input file")
+    parser.add_argument("-o", "--outputs", help="path to the .npz output file")
+    parser.add_argument(
+        "--print-time", action="store_true", help="record and print the execution time(s)"
+    )
+    parser.add_argument(
+        "--print-top",
+        metavar="N",
+        type=int,
+        help="print the top n values and indices of the output tensor",
+    )
+    parser.add_argument(
+        "--profile",
+        action="store_true",
+        help="generate profiling data from the runtime execution. "
+        "Using --profile requires the Graph Runtime Debug enabled on TVM. "
+        "Profiling may also have an impact on inference time, "
+        "making it take longer to be generated.",
+    )
+    parser.add_argument(
+        "--repeat", metavar="N", type=int, default=1, help="repeat the run n times. Defaults to '1'"
+    )
+    parser.add_argument(
+        "--rpc-key",
+        nargs=1,
+        help="the RPC tracker key of the target device",
+    )
+    parser.add_argument(
+        "--rpc-tracker",
+        nargs=1,
+        help="hostname (required) and port (optional, defaults to 9090) of the RPC tracker, "
+        "e.g. '192.168.0.100:9999'",
+    )
+    parser.add_argument("FILE", help="path to the compiled module file")
```

Comment:
how about change to `—file`? Other arguments starts `—`

Reply:
This is mostly to keep he convention other `tvmc` subcommands are using. It was more like a design decision, to mark the artefact in which the subcommand is acting on. Other tools, for example `clang` and `gcc` use similar convention.

(please note that I don't want to pretend being a specialist in command line tools, and I'm very happy with all the feeback I've been receiving :slightly_smiling_face:  )

Reply:
The argument without dashes are positional arguments. This behavior is same as most other compilers like `gcc <options> FILE`.

Reply:
thanks for explainment, this is better.